lib/sentry/opentelemetry/span_storage.ex

```diff
@@ -0,0 +1,151 @@
+defmodule Sentry.OpenTelemetry.SpanStorage do
+  @moduledoc false
+  use GenServer
+
+  defstruct [:cleanup_interval, :table_name]
+
+  alias Sentry.OpenTelemetry.SpanRecord
+
+  @cleanup_interval :timer.minutes(5)
+  @span_ttl :timer.minutes(30)
+
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(opts) when is_list(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  @impl true
+  def init(opts) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
+    cleanup_interval = Keyword.get(opts, :cleanup_interval, @cleanup_interval)
+
+    _ = :ets.new(table_name, [:named_table, :public, :ordered_set])
+
+    schedule_cleanup(cleanup_interval)
+
+    {:ok, %__MODULE__{cleanup_interval: cleanup_interval, table_name: table_name}}
+  end
+
+  @impl true
+  def handle_info(:cleanup_stale_spans, state) do
+    cleanup_stale_spans(state.table_name)
+    schedule_cleanup(state.cleanup_interval)
+
+    {:noreply, state}
+  end
```

Comment:
Nit, but can we group public API and callbacks together? Usual flow of a GenServer is

1. `start_link`
2. Public API (`store_span/2` and friends)
3. State (`defstruct`)
4. Callbacks
5.